packages/orchestration/src/utils/packet.js

```diff
@@ -0,0 +1,138 @@
+// @ts-check
+import { TxBody } from '@agoric/cosmic-proto/cosmos/tx/v1beta1/tx.js';
+import { RequestQuery } from '@agoric/cosmic-proto/tendermint/abci/types.js';
+import { atob, decodeBase64, encodeBase64 } from '@endo/base64';
+import {
+  CosmosQuery,
+  CosmosResponse,
+} from '@agoric/cosmic-proto/icq/v1/packet.js';
+
+/** @import { ResponseQuery } from '@agoric/cosmic-proto/tendermint/abci/types.js'; */
+
+/** @typedef {{ typeUrl: string; value: string; }} Proto3Msg */
+/** @typedef {{ path: string; data: string; height?: bigint; prove?: boolean; }} QueryMsg like RequestQuery from Tendermint ABCI, but with a base64 encoded `data` key */
+/** @typedef {Omit<ResponseQuery, 'key'|'value'> & { key: string; value: string; }} QueryResponse like ResponseQuery from Tendermint ABCI, but with a base64 encoded `key` and `value` keys */
```

Comment:
Let's try to hew closely to the external typedefs. 
```suggestion
/** @typedef {RequestQuery & { data: string; }} QueryMsg like RequestQuery from Tendermint ABCI, but with a base64 encoded `data` key */
/** @typedef {ResponseQuery & { key: string; value: string; }} QueryResponse like ResponseQuery from Tendermint ABCI, but with a base64 encoded `key` and `value` keys */
```

That reveals an undocumented difference: this had made `height` and `prove` optional but they aren't in `RequestQuery`:
```js
export interface RequestQuery {
    data: Uint8Array;
    path: string;
    height: bigint;
    prove: boolean;
}
```

Future improvements:
- a helper to generate a base64 type from interface
- update the codegen to make that the return type instead of `unknown`

Here's a helper we could use,
```
// TODO make codegen toJSON() return these instead of unknown
/**
 * Proto Any with arrays encoded as base64
 */
export type Base64Any<T> = {
  [Prop in keyof T]: T[Prop] extends Uint8Array ? string : T[Prop];
};
```
I'll try to incorporate that into https://github.com/Agoric/agoric-sdk/pull/9219/

Reply:
Done. I think we should try to land https://github.com/Agoric/agoric-sdk/pull/9219 first

Reply:
I pushed default values for `height` and `prove` out of this implementation, but we might consider including them in the future. The default values we get from `.fromPartial()` (`height: 0n, prove: false`)  are documented in the tests.

I'm not sure if these defaults are sensible, but responses seem to be reactive to state changes despite the `height: 0n` parameter. Perhaps, this parameter is more intended for historical queries.

`prove` is something we may want to investigate more - the ~only mention in [`cosmos/ibc-apps/modules/async-icq`](https://github.com/cosmos/ibc-apps/tree/18248c35e913b3ccba99a2756612b1fcb3370a0d/modules/async-icq) is a failing test which leads me to believe query proofs may not yet be supported:
https://github.com/cosmos/ibc-apps/blob/18248c35e913b3ccba99a2756612b1fcb3370a0d/modules/async-icq/keeper/relay_test.go#L132-L154